saq/queue/postgres.py

```diff
@@ -0,0 +1,521 @@
+
+            result = await cursor.fetchone()
+            assert result
+            return result[0]
+
+    async def schedule(self, lock: int = 1) -> t.Any: ...
+
+    async def sweep(self, lock: int = 60, abort: float = 5.0) -> list[t.Any]:
+        """Delete jobs past their TTL from the jobs table"""
+        async with self.pool.connection() as conn, conn.cursor() as cursor:
+            await cursor.execute(
+                f"""
+                DELETE FROM {self.jobs_table}
+                WHERE status IN ('aborted', 'complete', 'failed')
+                  AND %(now)s >= ttl
+                RETURNING key
+                """,
+                {"now": math.ceil(seconds(now()))},
+            )
+            await cursor.execute(
+                f"""
+                SELECT job
+                FROM {self.jobs_table} LEFT OUTER JOIN pg_locks ON lock_id = objid
+                WHERE status = 'active'
+                  AND objid IS NULL
+                """
+            )
+            results = await cursor.fetchall()
+        swept = []
+        for result in results:
+            job = self.deserialize(result[0])
+            if not job:
+                continue
+            swept.append(job.key)
+            if job.retryable:
+                await self.retry(job, error="swept")
+            else:
+                await job.finish(Status.ABORTED, error="swept")
+        return swept
+
+    async def listen(
+        self,
+        job_keys: Iterable[str],
+        callback: ListenCallback,
+        timeout: float | None = 10,
+    ) -> None:
+        if not job_keys:
+            return
+
+        async with self.pool.connection() as conn:
+            for key in job_keys:
+                await conn.execute(f'LISTEN "{key}"')
+            await conn.commit()
+            gen = conn.notifies(timeout=timeout or None)
+            async for notify in gen:
+                payload = self._load(notify.payload)
+                key = payload["key"]
+                status = Status[payload["status"].upper()]
+                if asyncio.iscoroutinefunction(callback):
+                    stop = await callback(key, status)
+                else:
+                    stop = callback(key, status)
+
+                if stop:
+                    await gen.aclose()
+
+    async def notify(self, job: Job) -> None:
+        payload = self._dump({"key": job.key, "status": job.status})
+        await self._notify(job.key, payload)
+
+    async def _notify(self, channel: str, payload: t.Any) -> None:
+        async with self.pool.connection() as conn:
+            await conn.execute(f"NOTIFY \"{channel}\", '{payload}'")
+
+    async def update(self, job: Job) -> None:
+        job.touched = now()
+        async with self._get_connection(job.key) as conn:
+            await conn.execute(
+                f"""
+                    UPDATE {self.jobs_table} SET job = %(job)s, status = %(status)s
```

Comment:
are you unable to pass in self.jobs_table as a param? what if thehre's sql injection in there